### ISSUE TITLE
save images / movies

### ISSUE THREAD

**Issue by ghorn** [ghorn/not-gloss#11]
http://hackage.haskell.org/package/OpenGL-2.11.1.0/docs/Graphics-Rendering-OpenGL-GL-ReadCopyPixels.html#v:readPixels

`screenShot :: VisObject a -> IO SomeBMPFormat`

then for movies

`movie :: [SomeBMPFormat] -> IO SomeMovieFormat`

**Comment by ghorn** [comment]
```haskell
import qualified Graphics.UI.GLUT as GLUT
import Graphics.Rendering.OpenGL.Raw
import Graphics.Rendering.OpenGL.GL.ReadCopyPixels ( readPixels )
import Graphics.Rendering.OpenGL ( PixelData(..), PixelFormat(RGB), DataType(UnsignedByte), get )
import Graphics.UI.GLUT.State ( screenSize )

import Foreign.Ptr
import Foreign.Marshal.Alloc
import Foreign.Marshal.Array

foo :: IO ()
foo = do
  size@(Size x y) <- get screenSize
  let pos = Position 0 0
  ptr <- mallocArray (fromIntegral (3*x*y)) :: IO (Ptr GLubyte)
  let pixelData = PixelData RGB UnsignedByte ptr
  readPixels pos size pixelData
  free ptr
```

**Comment by ghorn** [comment]
closed with 8eea7f5e2b6d7b7a322a07c894e234b0fb436ee6
